### PR TITLE
Fixed the interactive truck reset stuttering

### PR DESCRIPTION
**Pull request by ulteq**
Boom shakalaka ...

### CODE REVIEW

**Comment by ulteq** [comment]
Doesn't work either :disappointed: ... segfaults on truck deletion

**Comment by only-a-ptr** [comment]
Oh no no NO! :laughing: I don't want any more magic `//Must happen before|after|don't touch, the author knew what he was doing but he won't tell!` comments. The design is:
```
SyncWithSimThread()
UpdateSimulation() //  do all the legacy input/networking/graphics stuff that needs to be done
BufferSimData() // just before starting sim tasks, only then you have up-to-date simulation state
UpdateActors() // proceed!
```

Now that the steam is off, I'll actually read the commits now and figure out why it has to be this way

**Comment by only-a-ptr** [comment]
I wrote this to get us on the same page: https://github.com/RigsOfRods/rigs-of-rods/issues/1574

> The problem is that UpdateInputEvents() does sim tasks ...

It does modify the simulation state, that's why it needs to be done before SimBuffers (in principle). We may need some workarounds now, but those should be explicitly commented as such.

> And when do you update the new GfxActors?

Just left that out from the schematic, updated.

File: source/main/gfx/DustManager.cpp [hunk 1]
```diff
@@ -315,7 +315,17 @@ void RoR::GfxScene::RemoveGfxActor(RoR::GfxActor* remove_me)
         if (*itor == remove_me)
         {
             m_all_gfx_actors.erase(itor);
-            return;
+            break;
+        }
+    }
+    itor = m_live_gfx_actors.begin();
```
Comment (only-a-ptr) [inline source/main/gfx/DustManager.cpp]: needs a comment `//TODO:  this is a temporary workaround for actor removal being done when it shouldn't (legacy logic).`

**Comment by ulteq** [comment]
> It does modify the simulation state, that's why it needs to be done before SimBuffers

I think I see what you did there. I'll try to create a cleaner patch.

**Pull request closed without merging**